### ISSUE TITLE
Support opening app from template widget

### ISSUE THREAD

**Issue by potchin** [home-assistant/android#4067]
**Is your feature request related to a problem? Please describe.**
I have a text-based widget on my home screen which shows states of various HA entities. Frequently I'll want to open HA after seeing a state in the widget (eg I left a light on), it would be great to be able to tap the widget itself to launch the HA app.

At the moment the only time the widget supports a tap action is when there has been an error, which causes a retry. I suggest this is retained and the new functionality is only applied when the widget is in a non-error state.

**Describe the solution you'd like**
To be able to open the HA app by clicking on the text widget.

**Describe alternatives you've considered, if any**

**Additional context**

**Comment by dshokouhi** [comment]
> At the moment the only time the widget supports a tap action is when there has been an error, which causes a retry. I suggest this is retained and the new functionality is only applied when the widget is in a non-error state.

thats incorrect it also performs an update when there is no error shown to the user.
